frontend/NextJs/style-guide.md

```diff
@@ -0,0 +1,219 @@
+# Next.js Style Guide
+
+## Table of Contents
+
+1. [Basic Rules](#basic-rules)
+1. [Folder Structure](#folder-structure)
+1. [Fetching Data](#fetching-data)
+1. [Api Routes](#naming)
+1. [SEO](#mixins)
+
+## Basic Rules
+
+- If you are not using our Next.js Boostrap, make sure your the next.config.js
+  file include pageExtensions as follows
+  ```js
+  module.exports = {
+    pageExtensions: ['page.tsx', 'api.ts'],
+  }
+  ```
+- Only include one index.page.tsx component per page.
+- Always use JSX syntax.
+
+## Folder Structure
+
+```
+└───locales //folder to i18n per page
+│ └─── es
+│     └─── home.json
+│     └─── myscreenpage.json
+│     └───etc
+│ └─── en
+│     └─── home.json
+│     └─── myscreenpage.json
+│     └───etc
+│
+└───components
+│ └───baseComponents
+│     └───Input
+│     └───Text
+│     └───Button
+│     └───etc
+│
+└───pages
+│   | index.page.tsx // root or hoem page.
+│   └───api
+│   │   └───MyServerlessApi
+│   │       | index.api.tsx
+│   └───MyScreenPage
+│       | components/
+│       | constants.ts
+│       | index.page.tsx
+│       | styles.module.scss //only if you are using css modules.
+│       | utils.ts
+│       | assets // Screen specific app assets
+│
+└─── public
+│ └─── assets
+|
+└───config
+    | api.ts
+└───constants
+└─── store (Redux/Contex/Recoil)
+│   | store.ts
+│   └───myReducer
+│       | actions.ts
+│       | reducer.ts
+│       | selectors.ts
+└─── hooks
+└─── types
+└─── services
+    | MyService.ts
+└───utils
+│   index.ts
+```
+
+## Fetching Data
+
+Next.js allows you to fetch data in different ways, depending on your needs and
+preferences.
+
+- **getStaticProps**: The getStaticProps function is executed on the server side
+  is compiled, and allows you to retrieve data from an external API or any other
+  source and pass it as properties to the component. This function is util when
```

Comment:
I think that "util" as an adjective does not exist in english (idk if "util" may be used as a noun such as the "utils" directory on some projects though)
```suggestion
  source and pass it as properties to the component. This function is useful when
```